File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int maxLong = 0;
        
        Set<Integer> set = new HashSet<Integer>();
        for (int i = 0; i < nums.length; i++){
            set.add(nums[i]);
        }
        
        for (int i = 0; i < nums.length; i++){
            int curLong = 1;
            
            int num = nums[i];
            int j = 1;
                
            while (set.contains(num - j)){
                curLong ++;
                set.remove(num-j);
                j++;
            }
            
            j = 1;
            
            while (set.contains(num + j)){
                curLong ++;
                set.remove(num + j);
                j++;
            }
            
            
            maxLong = Math.max(maxLong, curLong);
        }
        
        return maxLong;
        
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        
        
        s = set(nums)
        
        for i, num in enumerate(nums):
            curLong = 1
            
            j = 1
            while (num+j) in s:
                s.remove(num+j)
                j += 1
                curLong += 1
            
            j = 1
            
            while (num-j) in s:
                s.remove(num-j)
                j += 1
                curLong += 1
            
            res = max(curLong, res)
        
        return res
            
            
            
        